extensions/interactions/NumberWithUnits/directives/number-with-units-rules.service.ts

```diff
@@ -50,11 +50,20 @@ export class NumberWithUnitsRulesService {
     var answerString = answerObject.toMathjsCompatibleString();
     var inputsString = inputsObject.toMathjsCompatibleString();
 
-    var answerList = this.unitsObjectFactory.fromRawInputString(
-      answerString).toDict();
-    var inputsList = this.unitsObjectFactory.fromRawInputString(
-      inputsString).toDict();
-    return this.utilsService.isEquivalent(answerList, inputsList);
+    var answerList: Partial<NumberWithUnitsAnswer> = this.unitsObjectFactory
+      .fromRawInputString(
+        answerString).toDict();
+    var inputsList: Partial<NumberWithUnitsAnswer> = this.unitsObjectFactory
+      .fromRawInputString(
+        inputsString).toDict();
+
+    delete answerList.units;
+    delete inputsList.units;
+
+    return (
+      this.IsEquivalentTo(answer, inputs) &&
```

Comment:
@Nik-09, that is a valid case and it is already being handled by the IsEquivalent function (it does the same thing as you suggested in this [comment](https://github.com/oppia/oppia/pull/19706/files#r1492764890)). but it does not check for strict value . So I used the utilsService.isEquivalent function without units, just for strict value comparison.

Here, the correct answer is 5 J/s, and the value is the same but the unit is different, hence getting rejected.

[303825277-e95a6af5-74fe-445b-8eaa-3f0b72293a14.webm](https://github.com/oppia/oppia/assets/79896602/9f93bebd-4ea3-45bb-850b-8b9049ccda84)

Reply:
Thanks for the explanation, sounds reasonable.